From 696cb6b4c86105498239cf4a571cbf48a6c21275 Mon Sep 17 00:00:00 2001
From: Vignesh Goutham Ganesh <vgg@amazon.com>
Date: Tue, 11 Jan 2022 21:05:13 -0800
Subject: [PATCH 01/16] OVA improvements

- Create /etc/pki/tls/certs dir as part of image-builds
- Tweak Product info in OVF
- Delete the VM after the build is complete

Signed-off-by: Vignesh Goutham Ganesh <vgg@amazon.com>
---
 images/capi/ansible/roles/sysprep/tasks/main.yml |  9 +++++++++
 images/capi/hack/ovf_template.xml                | 10 ++--------
 images/capi/packer/ova/packer-node.json          |  2 +-
 3 files changed, 12 insertions(+), 9 deletions(-)

diff --git a/images/capi/ansible/roles/sysprep/tasks/main.yml b/images/capi/ansible/roles/sysprep/tasks/main.yml
index 101a975b6..d7b33f803 100644
--- a/images/capi/ansible/roles/sysprep/tasks/main.yml
+++ b/images/capi/ansible/roles/sysprep/tasks/main.yml
@@ -27,6 +27,15 @@
 - ansible.builtin.import_tasks: photon.yml
   when: ansible_os_family == "VMware Photon OS"
 
+- name: Get stats of the /etc/pki folder
+  ansible.builtin.stat:
+    path: /etc/pki
+  register: pki
+
+- name: Ensures /etc/pki/tls/certs dir exists
+  file: path=/etc/pki/tls/certs state=directory
+  when: pki.stat.isdir is defined and pki.stat.isdir
+
 - name: Remove containerd http proxy conf file if needed
   ansible.builtin.file:
     path: /etc/systemd/system/containerd.service.d/http-proxy.conf
diff --git a/images/capi/hack/ovf_template.xml b/images/capi/hack/ovf_template.xml
index 316427ec3..ca23db5f9 100644
--- a/images/capi/hack/ovf_template.xml
+++ b/images/capi/hack/ovf_template.xml
@@ -147,19 +147,13 @@
     <vmw:BootOrderSection vmw:instanceId="8" vmw:type="disk">
       <Info>Virtual hardware device boot order</Info>
     </vmw:BootOrderSection>
-    <EulaSection>
-      <Info>An end-user license agreement</Info>
-      <License>
-${EULA}
-      </License>
-    </EulaSection>
     <ProductSection>
       <Info>Information about the installed software</Info>
       <Product>${PRODUCT}</Product>
-      <Vendor>VMware Inc.</Vendor>
+      <Vendor>AWS</Vendor>
       <Version>${TYPED_VERSION}</Version>
       <FullVersion>${TYPED_VERSION}</FullVersion>
-      <VendorUrl>https://vmware.com</VendorUrl>
+      <VendorUrl>https://aws.amazon.com/</VendorUrl>
       <Category>Cluster API Provider (CAPI)</Category>
       <Property ovf:userConfigurable="false" ovf:value="${BUILD_TIMESTAMP}" ovf:type="string" ovf:key="BUILD_TIMESTAMP"/>
       <Property ovf:userConfigurable="false" ovf:value="${BUILD_DATE}" ovf:type="string" ovf:key="BUILD_DATE"/>
diff --git a/images/capi/packer/ova/packer-node.json b/images/capi/packer/ova/packer-node.json
index a405aa16b..7105fd1f8 100644
--- a/images/capi/packer/ova/packer-node.json
+++ b/images/capi/packer/ova/packer-node.json
@@ -486,7 +486,7 @@
     "crictl_url": "https://github.com/kubernetes-sigs/cri-tools/releases/download/v{{user `crictl_version`}}/crictl-v{{user `crictl_version`}}-linux-amd64.tar.gz",
     "crictl_version": null,
     "datastore": "",
-    "destroy": "false",
+    "destroy": "true",
     "disk_size": "20480",
     "existing_ansible_ssh_args": "{{env `ANSIBLE_SSH_ARGS`}}",
     "export_manifest": "none",
-- 
2.49.0

